### DISCUSSION TITLE
different results from find_transform and estimate_transform

### DISCUSSION THREAD

**Discussion by yas-nakajima** [quatrope/astroalign#73] [Help Desk]
I expected that the followings give the same results, but a bit different, although it is acceptable. 
Do find_transform and estimate_transform use different clipping algorithms or parameters? 

```
[IN]
imdata1 = fits.getdata('tSUPA00215715.fits')
imdata2 = fits.getdata('tSUPA00215725.fits')

transf, (src, dst) = aa.find_transform(imdata2, imdata1)
print(transf.params)

tform = aa.estimate_transform('affine', src, dst)
print(tform.params)

[OUT]
[[ 9.99907916e-01  1.05148612e-05 -2.51020994e+01]
 [-1.05148612e-05  9.99907916e-01 -4.92418919e+01]
 [ 0.00000000e+00  0.00000000e+00  1.00000000e+00]]

[[ 9.99956673e-01 -2.08512775e-05 -2.51015953e+01]
 [-7.94858443e-06  9.99872371e-01 -4.92018502e+01]
 [ 0.00000000e+00  0.00000000e+00  1.00000000e+00]]
```

For this example, repeated executions give the same results. 
So I don't think RANSAC's nature accounts for it. 

Thanks.

**Comment by martinberoiz** [comment]
I honestly don't know much about endian issues and the FITS standard to know exactly what's going on, but this trick fixed the issue:

    from astropy.io import fits

    imdata1 = fits.getdata('tSUPA00215715.fits')
    imdata2 = fits.getdata('tSUPA00215725.fits')

    imdata1_fixed = np.array(imdata1, dtype="<f4")
    imdata2_fixed = np.array(imdata2, dtype="<f4")

    reg_im, footprint = aa.register(imdata1_fixed, imdata2_fixed)

I'll look more into this issue, but for now that should be a quick fix.

**Comment by yas-nakajima** [comment]
Thank you for your response. 

By the way, this is a new issue and your reply is for the previous issue. 

Best Regards

**Comment by yas-nakajima** [comment]
Just for your information. I don't think it is a bug of astroalign.  

```
[IN]: 
imdata1 = fits.getdata('tSUPA00215715.fits')
imdata1.dtype

[OUT]:
dtype('>f4')
```

The dtype is already 32-bit float before executing np.array(imdata1, dtype="<f4"), but it needs to be "converted" to '<f4' to run aa.register. 
It is strange and interesting. 
Possibly a bug of astropy.io.fits?

**Comment by martinberoiz** [comment]
Thanks for pointing that out. I've been thinking last weekend and this might be related to the [FITS standard](https://fits.gsfc.nasa.gov/fits_standard.html).

This is what they say in the PDF file:

> The individual data values shall be stored in big-endian byte order such that the byte containing the most significant bits of the value appears first in the FITS file, followed by the remaining bytes, if any, in decreasing order of significance.

But most computers are actually in [little endian](https://en.wikipedia.org/wiki/Endianness).

> The Intel x86 and AMD64 / x86-64 series of processors use the little-endian format.

I don't really know if the creators of FITS files take this into consideration when dumping binary data to the file (or if they should care). I'm not sure if astropy should flip the endianness of arrays to match the cpu either.

**Comment by martinberoiz** [comment]
I changed this into a discussion because I want to dig deeper into this issue.

**Reply by yas-nakajima** [reply]
Okay. I got it.
